File: qcs_bluestream_customs/public/quality_inspection.js
// Copyright (c) 2024, Quark Cyber Systems FZC
// For license information, please see license.txt


frappe.ui.form.on('Quality Inspection', {
    onload(frm) {
        frm.toggle_reqd([
            "reference_type",
            "reference_name",
        ], false);
    },
    inspection_type(frm) {
        if ( frm.doc.inspection_type == 'Incoming' ) {
            frm.toggle_reqd([
                "reference_type",
                "reference_name",
            ], true);     
        } else {
            frm.toggle_reqd([
                "reference_type",
                "reference_name",
            ], false);  
        }
    },
	validate(frm) {
		$.each(frm.doc.readings || [], function(i, d) {
			if(!d.manual_inspection) d.manual_inspection = frm.doc.manual_inspection;
		});
		refresh_field("readings");
	}
});

File: qcs_bluestream_customs/controller/quality_inspection.py
# Copyright (c) 2024, Quark Cyber Systems FZC
# For license information, please see license.txt


import frappe


def update_custom_quality_inspection_details(quality_inspection, method=None):
    if quality_inspection.reference_type == 'Job Card':
        job_card = frappe.get_doc('Job Card', quality_inspection.reference_name)
        work_order = frappe.get_doc('Work Order', job_card.work_order)
        sales_order = frappe.get_doc('Sales Order', work_order.sales_order)
        if quality_inspection.custom_operation != job_card.operation:
            quality_inspection.custom_operation = job_card.operation
        if quality_inspection.custom_work_order != job_card.work_order:
            quality_inspection.custom_work_order = job_card.work_order
        if work_order.sales_order:
            if quality_inspection.custom_sales_order != work_order.sales_order:
                quality_inspection.custom_sales_order = work_order.sales_order
            if quality_inspection.custom_customer != sales_order.customer:
                quality_inspection.custom_customer = sales_order.customer
        else:
            pass
    else:
        pass
